Supports adding Integration to `VendorText` on the web client

iobtl opened the pull request:
# Overview

There are two main changes this PR brings, both of which support the changes made on the web client: 

1. Completion of the Google OAuth authorization flow
- This is mainly achieved through `GoogleSheetsService`, which now stores a `HashMap<Uuid, GoogleSheetsToken>` mapping an `Integration` UUID to an access token. For now, the concept of an `Integration` has been rebranded to that of a single "Connection" between a user and Google. Much of the logic for the exchanging of the access token has been brought here:
- https://github.com/Torus-Portals/portals-backend/blob/976cbafc36ce6d2acbf23dae33081d05ab35aa3d/src/services/google_sheets_service.rs#L177-L243

2. GQL routes/resolvers to support the Integration options presented on the front end.
- The resolvers by nature are reasonably straightforward -- each of them simply makes a call to `GoogleSheetsService` (naturally as well, since it holds the access tokens), which then handles the API calls to Google to retrieve the list of spreadsheets, sheet names, etc.
- The services making the calls can be found here:
- https://github.com/Torus-Portals/portals-backend/blob/976cbafc36ce6d2acbf23dae33081d05ab35aa3d/src/services/google_sheets_service.rs#L260-L383

# Further Changes

- Add refresh token logic (some Rust subtleties to wrangle with for now, to do with mutating a stored `Hashmap`)
- Decouple the `googleSheetsFetchSheetValue` Query (it perhaps shouldn't be a Query, since it creates a new Dimension + Cell for populating in the `VendorTextBlock`). Still thinking of how to achieve this simply.

iobtl committed:
Modify `Integration` data structure

1. Not necessary for `Integration` to store data regarding the API, i.e.
Google Sheets in this case. An `Integration` will only serve as a
"Connection" to the API.

2. Also, added created_at/by and updated_at/by fields to `Integration`.

iobtl committed:
Refactor `GoogleSheetsService` to store <IntegrationId, GoogleSheetsToken>

Since each `Integration` is associated with one "Connection", it makes
sense to associate the access_token with each IntegrationId.
Also, the `authorize_google_sheets` Mutation should create the new
`Integration` to obtain the IntegrationId to store the token with.

iobtl committed:
Add more GQL resolvers/routes that deal with calling Google Sheets API

Notable changes:
1. New Google Drive scope added to request for access token to allow for
querying of list of spreadsheets to be selected.
2. All the resolvers (e.g. list_spreadsheets, list_spreadsheets_sheet_names)
that are used in the front-end for displaying Integration options have
been added.

iobtl committed:
Modify GSheets API value-fetching call to create Cell and return Dimension.

For now, a `BasicTextCell` storing a `GoogleSheetsDimension` is initialized
through the `googleSheetsFetchSheetValue` query (probably need to decouple
into a Mutation).

The Dimension is then returned to the front-end to be stored in a
`VendorText` block.

iobtl committed:
Add new scope to fetch user's Gmail to use for Integration name.

brochington reviewed (commented):
Overall very nice! Left a couple notes, but nothing drastic.

iobtl committed:
Some refactoring and clean up of code

Namely:
1. Moved Google API endpoints into `config.ron` and reformat scope string
2. Remove some redundant commented-out code
3. Simplify deserializing of API responses

brochington reviewed (approved):
LGTM!
